services/web/server/src/simcore_service_webserver/login/_registration.py

```diff
@@ -276,9 +278,8 @@ async def check_and_consume_invitation(
     # database-type invitations
     if confirmation_token := await validate_confirmation_code(invitation_code, db, cfg):
         try:
-            invitation_data: InvitationData = _InvitationValidator.parse_obj(
-                confirmation_token
-            ).data
+            validated = _InvitationValidator.parse_obj(confirmation_token)
+            invitation_data: InvitationData = validated.data  # type: ignore[assignment]
```

Comment:
actually you should not have to do that. Json[InvitationData] is of type InvitationData as it is a wrapper. The problem is that you enforce the type here and mypy rightfully complains.